src/layup/convert.py

```diff
@@ -0,0 +1,92 @@
+import os
+import math
+from pathlib import Path
+from typing import Literal
+
+from concurrent.futures import ProcessPoolExecutor
+
+from layup.file_io.CSVReader import CSVDataReader
+from layup.file_io.HDF5Reader import HDF5DataReader
+from layup.file_io.file_output import write_csv, write_hdf5
+
+
+def process_chunk(data):
+    pass
+
+
+def convert(
+    input: str,
+    output_file_stem: str,
+    convert_to: Literal["BCART", "BCOM", "BKEP", "CART", "COM", "KEP"],
+    file_format: Literal["csv", "hdf5"] = "csv",
+    chunk_size: int = 10_000,
+    num_workers: int = -1,
+):
+    input_file = Path(input)
+    output_file = Path(f"{output_file_stem}.{file_format.lower()}")
+    output_directory = output_file.parent.resolve()
+
+    if num_workers < 0:
+        num_workers = os.cpu_count()
+
+    # Check that input file exists
+    if not input_file.exists():
+        raise FileNotFoundError(f"Input file {input_file} does not exist")
```

Comment:
we have raise errors here but elsewhere we have logging that could go to standard error or a log file. So we probably want to be consistent

Reply:
It might be okay to raise errors as this is for command line, but the bits that might be run in an more automated way like using the orbit fitting functions might be okay with the logging that could go stderror or logfile.